main.py

```diff
@@ -290,12 +287,13 @@ def status(message):
     while True:
         try:
             data=queue.peek(blocking=True)
-            LOGGER.debug("Publishing:"+str(data))
-            info=client.publish("prism/monitorA002",str(data) , qos=1)
+            data={k.decode():(v.decode() if isinstance(v, bytes) else v, u.decode()) for k,(v,u) in data.items()}
+            data=json.dumps(data)
+            info=client.publish("prism/"+cfg['mqtt']['uname'],data, qos=1)
```

Comment:
A couple of things:

1. With Python, it's safer to use the [`string.format`](https://docs.python.org/3/library/string.html#string-formatting) method when combining strings. For example:

```
"prism/{}".format(cfg['mqtt']['uname'])
```

instead of

```
"prism/"+cfg['mqtt']['uname']
```

Python raises an exception if the thing you are concatenating is a number (for example `"test" + 5`). `uname` will probably never be a number, but you never know so it is safer to use the format method. In fact, using `format` is the only way I combine strings now.

2. I think the topic should be something like this: `"prisms/{}/data".format(cfg['mqtt']['uname'])`

Reply:
Actually, the `uname` should not be provided in the configuration file. The hostname of the device should be the user name. I get the hostname in line 253. So you could write something like:

```
info = client.publish("prisms/{}/data".format(hostname), data, qos=1)
```